llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1230,46 +1233,59 @@ RISCVVLOptimizer::getMinimumVLForUser(MachineOperand &UserOp) {
   // Looking for an immediate or a register VL that isn't X0.
   assert((!VLOp.isReg() || VLOp.getReg() != RISCV::X0) &&
          "Did not expect X0 VL");
+
+  // If we know the demanded VL of UserMI, then we can reduce the VL it
+  // requires.
+  if (DemandedVLs.contains(&UserMI)) {
```

Comment:
Is it really possible to have a VLOp but not be part of DemandedVLs?

Reply:
I thought the same thing originally first, but then found out we have vcpop.m etc. which don't define any vector operands but use vector operands. So in that case we can't reduce the demanded VL since it's not elementwise

Reply:
I'm confused.  vcpop.m is not a supported instruction.  Thus, it's VL should never be reduced, and it shouldn't have an element in the map (not a candidate).  It can be a valid *user* of a supported instruction, but that seems fine?  We'd return the VL of the vcpop.m instruction?

Reply:
Oops, ignore my comment. When I wrote it I was thinking about my initial version of this patch when I computed the demandedVL for every instruction that defined a vector, not just candidates, which is why I was seeing vcpop.m etc.

In the latest version of the patch we still need might have candidate instructions that have aborted, i.e. returned std::nullopt in checkUsers, and so don't have an entry.